messaging/src/main/java/org/axonframework/eventhandling/processors/streaming/pooled/ClaimTask.java

```diff
@@ -107,9 +107,8 @@ protected CompletableFuture<Boolean> task() {
         List<Segment> segments = joinAndUnwrap(
                 unitOfWorkFactory.create()
                                  .executeWithResult((context) ->
-                                                            CompletableFuture.completedFuture(
-                                                                    tokenStore.fetchAvailableSegments(name)
-                                                            )
+                                                            tokenStore.fetchAvailableSegments(name,
+                                                                                              null)
```

Comment:
Here we create the UnitOfWork to run it in a transaction. The transaction it's coupled to the ProcessingContext, so it'd be good to pass the contex to the method. Same in other places. 
```suggestion
                                                            tokenStore.fetchAvailableSegments(name,
                                                                                              context)
```